### DISCUSSION TITLE
How do I limit/clamp the camera's position to my level/TileMap?

### DISCUSSION THREAD

**Discussion by ManicMarrc** [ramokz/phantom-camera#109] [Question / Help Wanted]
Exactly as the title says, how would I go about limiting the camera's position to the bounding box of my level?

**Comment by ramokz** [comment]
At the minute there's no official way of doing this, but think it's a very worthwhile addition.

Am not overly familiar with TileMaps, so would you be okay with making a [Feature proposal](https://github.com/ramokz/phantom-camera/issues/new?assignees=&labels=feature-proposal%2Cneeds-triage&projects=&template=feature_proposal.yml) describing any details or additional nice-to-haves. E.g. would it be ideal to limit the camera to a bounding box as well as having an optional margin to not be clamped exactly to the edge of the TileMap?

**Reply by ManicMarrc** [reply]
It would have to be someone else unfortunately. I don't really know too much on writing a proper feature proposal.

**Reply by ramokz** [reply]
I've written up a proposal (#111), but I don't want to make (too many) assumptions about what an ideal solution will be, as I'm relatively inexperienced with TileMaps. Feel free to chip in with any suggestions if you think there are other aspects to consider for it than the ones written in there.

**Comment by ramokz** [comment]
Closing the discussion post as this has now become a feature proposal instead #111
